tests/unit/test_estimator.py

```diff
@@ -2219,72 +2226,75 @@ def test_fit_verify_job_name(strftime, sagemaker_session):
+        ),
+        (
+            True,
+            SAGEMAKER_CONFIG_TRAINING_JOB_WITH_DEBUG_HOOK_CONFIG_AS_TRUE,
+            S3_OUTPUT_PATH_FROM_SESSION_S3_DEFAULT_CONFIG,
+        ),
+        (False, SAGEMAKER_CONFIG_TRAINING_JOB_WITH_DEBUG_HOOK_CONFIG_AS_TRUE, False),
+        (
+            HOOK_CONFIG,
+            SAGEMAKER_CONFIG_TRAINING_JOB_WITH_DEBUG_HOOK_CONFIG_AS_TRUE,
+            HOOK_CONFIG.s3_output_path,
+        ),
+        (
+            HOOK_CONFIG_WITHOUT_S3_PATH,
+            SAGEMAKER_CONFIG_TRAINING_JOB_WITH_DEBUG_HOOK_CONFIG_AS_TRUE,
+            S3_OUTPUT_PATH_FROM_SESSION_S3_DEFAULT_CONFIG,
+        ),
+    ],
+)
+def test_prepare_for_training_for_debugger_hook_config_value_combinations(
     sagemaker_session,
+    sagemaker_config,
+    debugger_hook_config_direct_input,
+    expected_debugger_hook_config_output,
 ):
-    """
-    Tests that the default_bucket from sagemaker_config is NOT injected into the debugger_hook_config.s3_output_path
-    when full debugger_hook_config is provided from direct input
-    """
-    sagemaker_session.sagemaker_config = SAGEMAKER_CONFIG_TRAINING_JOB
-    debugger_hook_config_input = DebuggerHookConfig(
-        hook_parameters={
-            "save_interval": "100",
-        },
-        s3_output_path="s3://mytestbucket/testpath/",
-    )
+    sagemaker_session.sagemaker_config = sagemaker_config
     fw = DummyFramework(
         entry_point=SCRIPT_PATH,
         role=ROLE,
         sagemaker_session=sagemaker_session,
         instance_count=INSTANCE_COUNT,
         instance_type=INSTANCE_TYPE,
-        debugger_hook_config=debugger_hook_config_input,
+        debugger_hook_config=debugger_hook_config_direct_input,
     )
     fw._prepare_for_training()
-    assert fw.debugger_hook_config.hook_parameters == debugger_hook_config_input.hook_parameters
-    assert fw.debugger_hook_config.s3_output_path == debugger_hook_config_input.s3_output_path
-
 
-def test_prepare_for_training_with_sagemaker_config_injection_for_debug_hook_config_provided_from_direct_input_as_true(
-    sagemaker_session,
-):
-    """
-    Tests that the default_bucket from sagemaker_config is injected into the debugger_hook_config.s3_output_path
-    when debugger_hook_config is provided from direct input as True
-    """
-    sagemaker_session.sagemaker_config = SAGEMAKER_CONFIG_TRAINING_JOB
-    fw = DummyFramework(
-        entry_point=SCRIPT_PATH,
-        role=ROLE,
-        sagemaker_session=sagemaker_session,
-        instance_count=INSTANCE_COUNT,
-        instance_type=INSTANCE_TYPE,
-        debugger_hook_config=True,
-    )
-    fw._prepare_for_training()
-    assert fw.debugger_hook_config.s3_output_path == f"s3://{sagemaker_session.default_bucket()}/"
+    if expected_debugger_hook_config_output is False:
+        assert fw.debugger_hook_config == expected_debugger_hook_config_output
+    else:
+        assert fw.debugger_hook_config.s3_output_path == expected_debugger_hook_config_output
```

Comment:
optional: might be even more straightforward to just have the entire debug hook config in the parameterization above? That way we could also validate that the rest of the config also matches, not just the s3 path.

Reply:
The comment makes sense but I wanted to make the validation more focused towards the defaults s3 path behavior. Since we already know resolve_value_from_config always takes the direct_input as final value whenever provided.